controllers/utils/hash.go

```diff
@@ -61,23 +61,23 @@ func CreateUniqueReplicationId(clusterFSIDs map[string]string) (string, error) {
 }
 
 func GenerateUniqueIdForMirrorPeer(mirrorPeer multiclusterv1alpha1.MirrorPeer, hasStorageClientRef bool) string {
-	var peerAccumulator []string
-
+	var checksum [20]byte
 	if hasStorageClientRef {
+		var peerAccumulator []string
 		for _, peer := range mirrorPeer.Spec.Items {
 			peerAccumulator = append(peerAccumulator, GetKey(peer.ClusterName, peer.StorageClusterRef.Name))
 		}
+		sort.Strings(peerAccumulator)
+		checksum = sha1.Sum([]byte(strings.Join(peerAccumulator, "-")))
 	} else {
+		var peerAccumulator string
 		for _, peer := range mirrorPeer.Spec.Items {
-			peerAccumulator = append(peerAccumulator, peer.ClusterName)
+			peerAccumulator += peer.ClusterName
 		}
-
+		checksum = sha1.Sum([]byte(peerAccumulator))
 	}
-
-	sort.Strings(peerAccumulator)
-
-	checksum := sha1.Sum([]byte(strings.Join(peerAccumulator, "-")))
-	return hex.EncodeToString(checksum[:])
+	// truncate to bucketGenerateName + "-" + first 12 (out of 20) byte representations of sha1 checksum
+	return hex.EncodeToString(checksum[:])[0 : len(BucketGenerateName)+1+12]
```

Comment:
is this wont provide any length error for the older upgrade case, because I dont see you adding 1+12 in else case

Reply:
both if and else will calculate the checksum in different ways
This checksum will return the first 12 characters in hex string.

Reply:
ack